backend/service/audit/setup.go

```diff
@@ -156,9 +159,22 @@ func (c *client) poll(interval time.Duration) {
 		}
 	}
 
+	lockID := c.storage.GetLockID(auditEventSinkLockId)
 	ticker := time.NewTicker(interval)
 	for {
 		<-ticker.C
-		readAndFanout()
+
+		isLocked, err := c.storage.AttemptLock(context.TODO(), lockID)
+		if err != nil {
+			c.logger.Error("Error while attempting to get lock", zap.Error(err))
+		}
+
+		if isLocked {
+			readAndFanout()
+			_, err := c.storage.ReleaseLock(context.TODO(), lockID)
```

Comment:
Wasn't immediately sure what i wanted to do with these context's, but maybe ill create a context with a deadline of 1s?

Reply:
i would pull the context from readAndFanout up, make it 10 seconds or something, and then just share it between all of these funcs